src/general/image_utilities.cpp

```diff
@@ -5642,7 +5643,7 @@ Extracts parameters from <magick_image> that matter for a Cmgui_image
 				*number_of_components = 3;
 			}
 		}
-		*number_of_bytes_per_component = magick_image->depth/8;
+        *number_of_bytes_per_component = (magick_image->depth == 0) ? 0 : std::max(1, int(magick_image->depth/8));
```

Comment:
I think this should be:
```
*number_of_bytes_per_component = (magick_image->depth + 7) / 8;
```
So e.g. 12 bits rounds up to 2 bytes.

Can also remove `#include <algorithm>`.

Reply:
[2f6db9f](https://github.com/cmlibs/zinc/pull/287/commits/2f6db9f07ad2f1d08c6df821757e2179009b3c64)